docs/source/troubleshooting.mdx

```diff
@@ -14,17 +14,118 @@ See the License for the specific language governing permissions and
+
+<Tip>
+
+Refer to the Performance [guide](performance) for more details about memory-saving techniques.
+
+</Tip>
+
+## Unable to load a saved TensorFlow model
+
+TensorFlow's [model.save](https://www.tensorflow.org/tutorials/keras/save_and_load#save_the_entire_model) method will save the entire model - architecture, weights, training configuration - in a single file. However, when you load the model file again, you may run into an error because 🤗 Transformers may not load all the TensorFlow-related objects in the model file. To avoid issues with saving and loading TensorFlow models, we recommend you:
+
+- Save the model weights as a `h5` file extension with [`model.save_weights`](https://www.tensorflow.org/tutorials/keras/save_and_load#save_the_entire_model) and then reload the model with [`~TFPreTrainedModel.from_pretrained`]:
+
+```py
+>>> from transformers import TFPreTrainedModel
+>>> from tensorflow import keras
+
+>>> model.save_weights("some_folder/tf_model.h5")
+>>> model = TFPreTrainedModel.from_pretrained("some_folder")
+```
+
+- Save the model with [`~TFPretrainedModel.save_pretrained`] and load it again with [`~TFPreTrainedModel.from_pretrained`]:
+
+```py
+>>> from transformers import TFPreTrainedModel
+
+>>> model.save_pretrained("path_to/model")
+>>> model = TFPreTrainedModel.from_pretrained("path_to/model")
+```
+
+## ImportError
+
+Another common error you may encounter, especially if it is a newly released model, is `ImportError`:
+
+```
+ImportError: cannot import name 'ImageGPTFeatureExtractor' from 'transformers' (unknown location)
+```
+
+For these error types, check to make sure you have the latest version of 🤗 Transformers installed to access the most recent models:
+
+```bash
+pip install transformers
+```
+
+## CUDA error: device-side assert triggered
+
+Sometimes you may run into a generic CUDA error about an error in the device code.
+
+```
+RuntimeError: CUDA error: device-side assert triggered
+```
+
+You should try to run the code on a CPU first to get a more descriptive error message. Add the following environment variable to the beginning of your code to get a better traceback:
+
+```py
+>>> import os
+
+>>> os.environ["CUDA_LAUNCH_BLOCKING"] = "1"
+```
+
+Often, the error occurs when there is a mismatch between the number of labels and outputs, which produces an `IndexError`. You can try to fix this by truncating sequences to the `max_source_length` supported by the model you're using. 
+
+If you are using a script, add the following argument to your script:
+
+```bash
+--max_source_length 512
+```
+
+If using a notebook, specify the maximum length in the tokenizer's `max_length` argument. You should also restart your notebook to flush the earlier CUDA assertions.
+
+```py
+>>> from transformers import T5Tokenizer
+
+>>> tokenizer = T5Tokenizer.from_pretrained("t5-small")
+>>> max_source_length = 512
+
+>>> encoded_input = tokenizer(sentences, padding=True, truncation=True, max_length=max_source_length)
+```
```

Comment:
I would remove this entirely as it will confuse any user in their own example/ a text classification or a token classification example.